x/epochs/types/hooks.go

```diff
@@ -0,0 +1,33 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type EpochHooks interface {
+	// the first block whose timestamp is after the duration is counted as the end of the epoch
+	AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64)
+	// new epoch is next block of epoch end block
+	BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochNumber int64)
+}
+
+var _ EpochHooks = MultiEpochHooks{}
+
+// combine multiple gamm hooks, all hook functions are run in array sequence
+type MultiEpochHooks []EpochHooks
+
+func NewMultiEpochHooks(hooks ...EpochHooks) MultiEpochHooks {
+	return hooks
+}
+
+func (h MultiEpochHooks) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) {
+	for i := range h {
+		h[i].AfterEpochEnd(ctx, epochIdentifier, epochNumber)
+	}
+}
+
+func (h MultiEpochHooks) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochNumber int64) {
```

Comment:
Is this the upcoming epoch number, or the number that just ended? Can the variable be renamed here + a comment added for that?

Reply:
I have added comment, if we rename var name, it would be longer.